File: app/src/main/java/com/example/bmicalculator/MainActivity.java
package com.example.bmicalculator;

import android.os.Bundle;
import android.view.MenuItem;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import com.google.android.material.bottomnavigation.BottomNavigationView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        BottomNavigationView bottomNav = findViewById(R.id.bottom_navigation);
        bottomNav.setOnNavigationItemSelectedListener(navigationListener);

        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container,
                    new StartFragment()).commit();
        }
    }

    private BottomNavigationView.OnNavigationItemSelectedListener navigationListener =
            new BottomNavigationView.OnNavigationItemSelectedListener() {
                @Override
                public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                    Fragment selectedFragment = null;

                    switch (item.getItemId()) {
                        case R.id.bmiNav:
                            selectedFragment = new BmiFragment();
                            break;
                        case R.id.caloriesNav:
                            selectedFragment = new CaloriesFragment();
                            break;
                        case R.id.foodNav:
                            selectedFragment = new FoodFragment();
                            break;
                        case R.id.quizNav:
                            selectedFragment = new QuizFragment();
                            break;
                        case R.id.statNav:
                            selectedFragment = new StatisticFragment();
                    }

                    getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container,
                            selectedFragment).commit();

                    return true;
                }
            };

}

File: app/src/main/java/com/example/bmicalculator/kotlinclasses/MainActivity.kt
package com.example.bmicalculator.kotlinclasses

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.example.bmicalculator.R
import com.google.android.material.bottomnavigation.BottomNavigationView


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val bottomNav = findViewById<BottomNavigationView>(R.id.bottom_navigation)
        bottomNav.setOnNavigationItemSelectedListener(navigationListener)

        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction().replace(R.id.fragment_container,
                    StartFragment()).commit()
        }
    }

    private val navigationListener = BottomNavigationView.OnNavigationItemSelectedListener { item ->

        var selectedFragment: Fragment? = null

        when (item.itemId) {
            R.id.bmiNav -> selectedFragment = BmiFragment()
            R.id.caloriesNav -> selectedFragment = CaloriesFragment()
            R.id.foodNav -> selectedFragment = FoodFragment()
            R.id.quizNav -> selectedFragment = QuizFragment()
            R.id.statNav -> selectedFragment = StatisticFragment()
        }

        supportFragmentManager.beginTransaction().replace(R.id.fragment_container,
                selectedFragment!!).commit()
        true
    }
}
